$NetBSD: patch-gio_tests_meson.build,v 1.1 2020/01/22 17:09:44 minskim Exp $

Darwin: Set LD_LIBRARY_PATH to run uninstalled executable with @rpath.

--- gio/tests/meson.build.orig	2019-12-19 16:33:15.000000000 +0000
+++ gio/tests/meson.build
@@ -470,7 +470,9 @@ if not meson.is_cross_build() or meson.h
   plugin_resources_c = custom_target('plugin-resources.c',
     input : 'test4.gresource.xml',
     output : 'plugin-resources.c',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--generate-source',
@@ -494,7 +496,9 @@ if not meson.is_cross_build() or meson.h
   test_gresource = custom_target('test.gresource',
     input : 'test.gresource.xml',
     output : 'test.gresource',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--sourcedir=' + meson.current_build_dir(),
@@ -505,7 +509,9 @@ if not meson.is_cross_build() or meson.h
   test_resources2_c = custom_target('test_resources2.c',
     input : 'test3.gresource.xml',
     output : 'test_resources2.c',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--generate',
@@ -516,7 +522,9 @@ if not meson.is_cross_build() or meson.h
   test_resources2_h = custom_target('test_resources2.h',
     input : 'test3.gresource.xml',
     output : 'test_resources2.h',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--generate',
@@ -528,7 +536,9 @@ if not meson.is_cross_build() or meson.h
     input : 'test2.gresource.xml',
     depends : big_test_resource,
     output : 'test_resources.c',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--sourcedir=' + meson.current_build_dir(),
@@ -539,7 +549,9 @@ if not meson.is_cross_build() or meson.h
   digit_test_resources_c = custom_target('digit_test_resources.c',
     input : '111_digit_test.gresource.xml',
     output : 'digit_test_resources.c',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--sourcedir=' + meson.current_build_dir(),
@@ -550,7 +562,9 @@ if not meson.is_cross_build() or meson.h
   digit_test_resources_h = custom_target('digit_test_resources.h',
     input : '111_digit_test.gresource.xml',
     output : 'digit_test_resources.h',
-    command : [glib_compile_resources,
+    command : [env_program.path(),
+               'LD_LIBRARY_PATH=gio:glib:gmodule:gobject',
+    	       glib_compile_resources,
                '--target=@OUTPUT@',
                '--sourcedir=' + meson.current_source_dir(),
                '--generate',
